test/inc/demo/type_erasure_shared_pointer_value_manager.hpp

```diff
@@ -57,27 +70,41 @@ struct UserValueManagement {
         }
 
         static void copyOp(void *to, const void *from) {
-            auto downcast = static_cast<const SPM *>(from);
-            new(to) SPM(*downcast);
+            auto downcast = static_cast<SPM *>(const_cast<void *>(from));
+            auto destValueManager = new(to) SPM;
+            *destValueManager->sharedPointer() = *downcast->sharedPointer();
         }
 
         constexpr static inline typename GP::VTable Operations = {
             AffordanceSpecifications::template Operation<SPM>...
         };
 
+        auto isExclusive() const noexcept {
+            auto sp = const_cast<SPM *>(this)->sharedPointer();
+            return 1 == sp->use_count();
+        }
+
+        auto makeExclusive() {
+            *sharedPointer() = std::make_shared<V>(*value());
```

Comment:
Not that this would be an issue for the vast majority of cases, but maybe const qualify the lvalue reference so it always selects the copy constructor.
```suggestion
            *sharedPointer() = std::make_shared<V>(static_cast<const V &>(*value()));
```